build.rs

```diff
@@ -49,49 +50,104 @@ fn main() {
             .build(),
 
         HostType::MacOS => {
-            let path_openssl = if cfg!(target_arch = "aarch64") {
-                "/opt/homebrew/Cellar/openssl@1.1"
-            } else {
-                "/usr/local/Cellar/openssl@1.1"
-            };
-            let mut openssl_dirs = std::fs::read_dir(PathBuf::new().join(path_openssl))
-                .unwrap()
-                .map(|r| r.unwrap().path())
-                .collect::<Vec<PathBuf>>();
-            openssl_dirs.sort_by(|a, b| {
-                let a_time = a.metadata().unwrap().modified().unwrap();
-                let b_time = b.metadata().unwrap().modified().unwrap();
-                b_time.cmp(&a_time)
-            });
-            let openssl_root = openssl_dirs[0].clone();
-            println!(
-                "cargo:rustc-link-search=native={}",
-                openssl_root.join("lib").display()
-            );
-            Config::new("mgclient")
-                .define("OPENSSL_ROOT_DIR", format!("{}", openssl_root.display()))
-                .define(
-                    "OPENSSL_CRYPTO_LIBRARY",
-                    format!(
-                        "{}",
-                        openssl_root.join("lib").join("libcrypto.dylib").display()
-                    ),
-                )
-                .define(
-                    "OPENSSL_SSL_LIBRARY",
-                    format!(
-                        "{}",
-                        openssl_root.join("lib").join("libssl.dylib").display()
-                    ),
+            println!("Checking for MacPorts.");
+            let output = Command::new("/usr/bin/command")
+                .args(["-v", "port"])
+                .output()
+                .expect("Failed to execute shell command: '/usr/bin/command -v port'")
+                .stdout;
+
+            let port_path = String::from_utf8(output).unwrap();
+            let port_path = &port_path[..port_path.len() - 1];
```

Comment:
Somehow `port_path` is empty here in my case
<img width="1717" alt="Screenshot 2022-11-10 at 8 22 19 PM" src="https://user-images.githubusercontent.com/4950251/201187477-41b0bb72-a207-41cd-acc5-f6b741348f7c.png">

Reply:
Sorry about that! I'll push a fix.

Reply:
@gitbuda  - Done.

Reply:
Works now, we'll merge shortly 😄

Reply:
Thanks @gitbuda !